src/app/problems/problem-view/problem-view.component.ts

```diff
@@ -41,4 +42,50 @@ export class ProblemViewComponent implements OnInit {
             this.user = user;
         });
     }
+
+    startTimer(uqj: number): void{
```

Comment:
Would there not need to be a function to end the timer? I'm a little confused about the behaviour of this function I feel like it can be simplified. Something similar to this solution may work well: https://stackoverflow.com/questions/43294235/angular-2-track-user-spent-time-on-a-page. Just something to consider and look into.

Reply:
I think it could be simplified but probably not much. Currently the timer runs when the page is focused and stops when user goes to another page. Since there isn't quite a 'start' button for each question, student might take a look at the question and start working on something else, and that's why this is kinda huge.

Reply:
Okay cool, I think that is a good implementation and it makes sense based on your desired use case. Definitely helps to understand the usage of local storage.